src/Process/Pool/Strategy.php

```diff
@@ -171,8 +171,12 @@ protected function _unPauseListenerProcesses(): Strategy
                 if (!$this->_getProcessPool()->isFull()) {
                     $typeCode = $listenerProcess->getTypeCode();
                     $newListenerProcess = $this->_getProcessCollection()->getProcessPrototypeClone($typeCode);
-                    while (!$this->_getProcessPool()->isFull() && $listenerProcess->hasMessages()) {
-                        $listenerProcess->processMessages();
+                    while (
+                        !$this->_getProcessPool()->isFull() &&
+                        $this->_getProcessPool()->canEnvironmentSustainAdditionProcesses() &&
```

Comment:
was `$this->_getProcessPool()->canEnvironmentSustainAdditionProcesses()` a part of the original condition? if not are we sure this is safe? meaning, was there a reason it was not?

Reply:
If I'm reading this correctly (which is unlikely), it looks like the process pool will get filled here if there is an excess of messages. Nothing further down the line will check if the exec environment can sustain more processes.

Say there's one exec environment, and one worker working a job. That worker is using up all the resources of the exec environment, but scheduling a ton of new jobs. Now the following happens:

1. The Selector sees there are a bunch of jobs waiting to be worked, and it publishes a bunch of messages to redis to add more processes
1. The Root's Listener gets one of those messages, and exits
1. Something(?) invokes `Process\Pool\Strategy::_unPauseListenerProcesses()`
1. `<process pool size> - 1` new processes are spawned without polling the exec environment state

Does that seem accurate to you?